package/samplers/hill_climb_search/hill_climb_search.py

```diff
@@ -0,0 +1,181 @@
+                final_neighbors.append(neighbor)
+
+        return final_neighbors
+
+    def _generate_neighbors(
+        self,
+        current_point: dict,
+        search_space: dict[str, optuna.distributions.BaseDistribution],
+        study: optuna.study.Study,
+    ) -> list[dict]:
+        """This function generates the neighbors of the current point"""
+        neighbors = []
+        for param_name, param_distribution in search_space.items():
+            if isinstance(param_distribution, optuna.distributions.FloatDistribution):
+                current_value = current_point[param_name]
+                step = param_distribution.step
+
+                neighbor_low = max(param_distribution.low, current_value - step)
+                neighbor_high = min(param_distribution.high, current_value + step)
+
+                neighbor_low_point = current_point.copy()
+                neighbor_low_point[param_name] = neighbor_low
+                neighbor_high_point = current_point.copy()
+                neighbor_high_point[param_name] = neighbor_high
+
+                neighbors.append(neighbor_low_point)
+                neighbors.append(neighbor_high_point)
+            else:
+                raise NotImplementedError
+
+        valid_neighbors = self._remove_tried_points(neighbors, study, current_point)
+
+        return valid_neighbors
+
+    def sample_relative(
+        self,
+        study: optuna.study.Study,
+        trial: optuna.trial.FrozenTrial,
+        search_space: dict[str, optuna.distributions.BaseDistribution],
+    ) -> dict[str, Any] | None:
+        if search_space == {}:
+            return {}
+
+        if self._current_state == "Not Initialized":
+            # Create the current point
+            starting_point = self._generate_random_point(search_space)
+            self._current_point = starting_point
+
+            # Add the neighbors
+            neighbors = self._generate_neighbors(starting_point, search_space, study)
+            self._remaining_points.extend(neighbors)
+
+            # Change the state to initialized
+            self._current_state = "Initialized"
+
+            # Return the current point
+            return starting_point
+
+        elif self._current_state == "Initialized":
+            # This section is only for storing the value of the current point and best neighbor point
+            previous_trial = study.get_trials(deepcopy=False)[-2]
+            if previous_trial.params == self._current_point:
+                # Just now the current point was evaluated
+                # Store the value of the current point
+                self._current_point_value = previous_trial.value
+            else:
+                # The neighbor was evaluated
+                # Store the value of the neighbor, if it improves upon the current point
+                neighbor_value = previous_trial.value
+
+                if neighbor_value < self._current_point_value:
```

Comment:
It seems that this implementation always assumes minimization, but the condition generally needs to consider the direction of the study. Because when maximization, it should be `neighbor_value > self._current_point_value`.

```python
study.create_study(..., direction="minimize")  # study.direction will be set to study.StudyDirection.MINIMIZE.
study.create_study(..., direction="maximize")  # study.direction will be set to study.StudyDirection.MAXIMIZE.
```

Please take a look at the documentation.
- [optuna.study.create_study](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.create_study.html#optuna.study.create_study)
  - > If none of direction and directions are specified, the direction of the study is set to “minimize”.
- [optuna.study.StudyDirection](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.StudyDirection.html#optuna.study.StudyDirection)

Reply:
Sure, thanks for the pointer, I have incorporated the same now.